types/braintree/index.d.ts

```diff
@@ -358,6 +358,36 @@ declare namespace braintree {
      */
 
     export class CreditCard {
+        static CardType: {
```

Comment:
I like making Card Type more specific. But have a few questions.

1. Should we remove the other `cardType` prop below?
2. Does this need to be lower-cased `cardType` to match docs? (https://developers.braintreepayments.com/reference/response/credit-card/node)
3. What is the purpose of `All`?

Similar questions with respect to CustomerLocation and CardTypeIndicator.

Reply:
Hi! Thanks for the review!

1. No, they're both necessary, the one I added is static and it works sorta like an enum. If you look at [this](https://github.com/braintree/braintree_node/blob/master/lib/braintree/credit_card.js) file in the source code of the library you'll notice that the value is being initialized in a `static initClass` method. The one I needed in my project at the moment was the `MerchantAccount.FundingDestination` "enum" but I saw many of them were not typed so I thought I might as well try to cover most, if not all of them, in my PR.

2. No, the example I was following was [this](https://developers.braintreepayments.com/guides/braintree-marketplace/onboarding/node). Here you'll notice when setting `funding.destination` it uses this `MerchantAccount.FundingDestination` enum, that's the same I wanted to achieve adding the other two `static` fields.

3. Looking at its implementation ([here](https://github.com/braintree/braintree_node/blob/master/lib/braintree/credit_card.js)) it looks like it returns an array with all the values in the object but itself, so in this case it would be `[CreditCard.CardType.AmEx, CreditCard.CardType.CarteBlanche, ...]`.

Sorry bout the lack of tests, I'll add them now. Same with any static field I might have missed.

Thanks again for the review! Let me know if there's anything else you'd like to discuss or if I didn't explain myself properly here!

Reply:
Nice, this is helpful. Didn't know about this!